platzky/engine.py

```
@@ -58,52 +121,206 @@ def __init__(
+        """Return all registered plugins of the given capability type."""
+        return self.plugins.get(plugin_type, [])
+
+    def get_plugin_infos(self) -> list[Any]:
+        """Return PluginInfo metadata for all loaded plugins."""
+        return [plugin.get_info() for plugin in self.loaded_plugins]
+
+    def notify(
+        self,
+        message: str,
+        topic: NotificationTopic = "general",
+        attachments: list[AttachmentProtocol] = [],
```

Ruff (0.15.12)

[warning] Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)